services/QuillLMS/client/app/bundles/Shared/hooks/__tests__/useModalAccessibility.test.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import useModalAccessibility from '../useModalAccessibility';
+
+// Mock Modal Component using the useModalAccessibility hook
+const MockModal = ({ onClose }) => {
+  const { modalRef } = useModalAccessibility(onClose);
+  return (
+    <div ref={modalRef} tabIndex={-1}>
+      <input aria-label="Input Field" />
+      <button type="button">Button</button>
+    </div>
+  );
+};
+
+describe('useModalAccessibility', () => {
```

Comment:
Just wanted to say how much I'm loving seeing these types of tests in the front-end.  It's so nice to feel like this functionality has test coverage.